cmd/agent/flow_run.go

```diff
@@ -0,0 +1,211 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"os/signal"
+	"sync"
+
+	"github.com/fatih/color"
+	"github.com/go-kit/log/level"
+	"github.com/gorilla/mux"
+	"github.com/grafana/agent/pkg/flow"
+	"github.com/grafana/agent/pkg/flow/logging"
+	"github.com/grafana/agent/pkg/river/diag"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/spf13/cobra"
+
+	// Install Components
+	_ "github.com/grafana/agent/component/all"
+)
+
+func runCommand() *cobra.Command {
+	r := &flowRun{
+		httpListenAddr:       "127.0.0.1:12345",
+		storagePath:          "data-agent/",
+		enableDebugEndpoints: true,
+	}
+
+	var cmd = &cobra.Command{
+		Use:   "run [flags] file",
+		Short: "Run Grafana Agent Flow",
+		Long: `The run subcommand runs Grafana Agent Flow in the foreground until an interrupt
+is received.
+
+run must be provided an argument pointing at the River file to use. If the
+River file wasn't specified, can't be loaded, or contains errors, run will exit
+immediately.
+
+run starts an HTTP server which can be used to debug Grafana Agent Flow or
+force it to reload (by POSTing to /-/reload). The listen address can be changed
```

Comment:
Nitpick, but we don't currently enforce the request to be a POST, which I think is okay.